plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/OpenShiftServerUtils.java

```diff
@@ -753,4 +753,10 @@ public static CoreException toCoreException(String msg) {
 		return collection;
 	}
 
+	public static void setModulePublishState(int state, IModule[] modules, Server server) {
```

Comment:
WTP has two different concepts of what an IModule[] array means. 
One is an array of module items, just a list basically, which may or may not have anything to do with each other. 
The second is a specific module and its parent modules in an array representing something like a tree-path to a single module. 

Your method signature or javadoc here should make clear whether you are setting the publish state of every module in the list (setModulesPublishStates) or a single module.